traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -416,12 +418,22 @@ func (cg *TOCacheGroup) createCoordinate() (*int, error) {
 
 func (cg *TOCacheGroup) updateCoordinate() error {
 	if cg.Latitude != nil && cg.Longitude != nil {
-		q := `UPDATE coordinate SET name = $1, latitude = $2, longitude = $3 WHERE id = (SELECT coordinate FROM cachegroup WHERE id = $4)`
-		result, err := cg.ReqInfo.Tx.Tx.Exec(q, tc.CachegroupCoordinateNamePrefix+*cg.Name, *cg.Latitude, *cg.Longitude, *cg.ID)
+		q := `SELECT * FROM coordinate WHERE id != (SELECT coordinate FROM cachegroup WHERE id = $1) AND name =$2`
```

Comment:
Instead of checking the number of rows returned by this query, why not use a query that returns the number of rows? E.g.
```go
const numberOfDuplicatesQuery = `
SELECT COUNT(*)
FROM public.coordinate
WHERE id NOT IN (
    SELECT coordinate
    FROM public.cachegroup
    WHERE id = $1
)
AND name = $2
`

func (cg *TOCacheGroup) updateCoordinate() error {
    if cg.Latitude != nil && cg.Longitude != nil {
        var count uint
        if err := cg.ReqInfo.Tx.Tx.QueryRow(numberOfDuplicatesQuery, *cg.ID, tc.CachegroupCoordinateNamePrefix+*cg.Name).Scan(&count); err != nil {
            return fmt.Errorf("getting coordinate for Cache Group '%s': %w", *cg.Name, err)
        }
        if count > 0 {
        // ... etc.
```

Reply:
also, query should specify `public` schema, as in my example

Reply:
updated.